src/main/java/com/stuypulse/robot/subsystems/Climber.java

```diff
@@ -5,24 +5,155 @@
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 /*-
  * Climbs at end of match
  *
  * Contains:
- *      - TBD
+ *      - Change tilt of climber
+ *      - Move climber via motor
+ *      - Different tilt angles
+ *      - Encoder + Solenoid used for stopping
  *
- * @author [TBD]
+ * @author independence106(Jason Zhou)
+ * @author Ca7Ac1(Ayan Chowdhury)
+ * @author marcjiang7(Marc Jiang)
+ * @author ambers7(Amber Shen)
+ * @author Souloutz(Howard Kong)
+ * @author jiayuyan0501(Jiayu Yan)
+ * @author ijiang05(Ian Jiang)
+ * @author TraceyLin (Tracey Lin)
+ * @author annazheng14(Anna Zheng)
+ * @author lonelydot(Raymond Zhang)
+ * @author andylin2004 (Andy Lin)
  */
 public class Climber extends SubsystemBase {
 
+    private Solenoid solenoidLong;
+    private Solenoid solenoidShort;
+
+    private DigitalInput limitSwitch;
+
+    private Solenoid stopper;
+
+    private RelativeEncoder encoder;
+
+    private CANSparkMax climber;
+
     public Climber() {
-        // Initialize Motors, etc... here!
+        climber = new CANSparkMax(Constants.Ports.Climber.MOTOR);
+        climber.setInverted(Constants.ClimberSettings.MOTOR_INVERTED);
+  
+        encoder = climber.getEncoder();
+        encoder.setPositionConversionFactor(Constants.ClimberSettings.CIRCUMFERENCE * Constants.ClimberSettings.GEAR_RATIO);
+
+        solenoidLong = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_LONG);
+        solenoidShort = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.SOLENOID_SHORT);
+        stopper = new Solenoid(PneumaticsModuleType.CTREPCM, Constants.Ports.Climber.STOPPER);
+
+        limitSwitch = new DigitalInput(Constants.Ports.Climber.LIMIT_SWITCH);
+    }
+    
+    private void moveMotor(double speed) {
+        climber.set(speed);
+    }
+
+    public void liftUp() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void liftDown() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_DEFAULT_SPEED.get());
+    }
+
+    public void stop() {
+        moveMotor(0.0);
+    }
+    
+    public void liftUpSlow() {
+        moveMotor(Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void liftDownSlow() {
+        moveMotor(-Constants.ClimberSettings.CLIMBER_SLOW_SPEED.get());
+    }
+
+    public void extendLong() {
+        solenoidLong.set(true);
+    }
+    
+    public void retractLong() {
+        solenoidLong.set(false);
+    }
+    
+    public void extendShort() {
+        solenoidShort.set(true);
+    }
+
+    private void retractShort() {
+        solenoidShort.set(false);
+    }
+
+    public void fullyRetract() {
+        retractShort();
+        retractLong();
+    }
+
+    public void fullyExtend() {
+        extendLong();
+        extendShort();
+    }
```

Comment:
IK sam said something in person about these methods but IDK what it was. if these methods are problematic though one alternative thing i can think of is:

* enum for climber configuration that stores the state of the two solenoids
* have one method that implements using these configurations 
* this enum can store only configurations that make sense